src/normalize/quotation.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Editor, NodeEntry, Transforms } from 'slate'
+import { List } from 'slate-lists'
+
+import { Paragraph, Quotation, Title } from '../interfaces'
+
+/**
+ * Normalize an quotation
+ *
+ * Return true if entry was normalized and false otherwise.
+ */
+export default function normalizeQuotation(editor: Editor, entry: NodeEntry): boolean {
+    const [node, path] = entry
+
+    if (Quotation.isQuotation(node)) {
+        // Quotation's may contain only Titles, Paragraphs,
+        // Lists and other Quotations
+        let index = 0
+        for (const child of node.children) {
+            if (
+                Title.isTitle(child)
+                || Paragraph.isParagraph(child)
+                || List.isList(child)
+                || Quotation.isQuotation(child)
+            ) {
+                index++
+                continue
+            }
+
+            Transforms.liftNodes(editor, { at: [...path, index] })
+            return true
```

Comment:
1. Remove empty quotations

   ```typescript
   // A quotation must not be empty.
   if (node.children.length === 0) {
       Transforms.removeNodes(editor, { at: path })
       return true
   }
   ```

2. I know eslint doesn't catch this, but please keep formatting of `if`s consistent.

3. Please use `normalizeOrderedChildren` instead of walking children manually. I know that right now there is no difference (since ensuring that `Title` must be the first child is done by `Title`'s normalizations), but when we add support for citations there might be, as we'll probably want quotation's source to always be last.

4. Unwrap nested quotation if it is not selected and it's the only child of another quotation.

Reply:
I added normalizations for 1 and 4 but it seems like tests for these cases are already there and they are passing without these normalization